debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerConnection.java

```diff
@@ -282,8 +282,10 @@ public void getChangesForTables(String databaseName, SqlServerChangeTable[] chan
         prepareQuery(queries, preparers, consumer);
     }
 
-    private Lsn getFromLsn(String databaseName, SqlServerChangeTable changeTable, Lsn intervalFromLsn) throws SQLException {
-        Lsn fromLsn = changeTable.getStartLsn().compareTo(intervalFromLsn) > 0 ? changeTable.getStartLsn() : intervalFromLsn;
+    private Lsn getFromLsn(String databaseName, SqlServerChangeTable changeTable, Lsn intervalFromLsn, Lsn intervalToLsn) throws SQLException {
+        Lsn fromLsn = changeTable.getStartLsn().compareTo(intervalFromLsn) > 0 && changeTable.getStartLsn().compareTo(intervalToLsn) < 0
```

Comment:
to ensure `fromLsn` is inside `[intervalFromLsn, intervalToLsn]`, sometimes it's not the case

Reply:
If `startLsn > fromLsn` and `startLsn >= toLsn`, it means that the table is outside of the interval. Should this even be possible?

I believe we should assume that if the change table was selected, there's an intersection between the LSNs of the changes it represents and the interval being streamed. Otherwise, why do we even consider it at this iteratilon?

Shouldn't it be just `max(startLsn, fromLsn)`? I.e. not taking the `toLsn` into account.